easybuild/framework/easyblock.py

```diff
@@ -4198,6 +4207,98 @@ def print_dry_run_note(loc, silent=True):
+            dirpath = create_unused_path(path)
+            persistent_dirs[dirname] = dirpath
+        return persistent_dirs[dirname]
+
+    def get_artifact_dir(path):
+        return get_dir(path, 'artifact')
+
+    def get_log_dir(path):
+        return get_dir(path, 'log')
+
+    def initialize_persistent_dir_and_replicate_if_paths_distinct(replication_operation,
+                                                                  get_directory,
+                                                                  source_path,
+                                                                  destination_base_path,
+                                                                  destination_relative_path):
+        abs_source_path = os.path.abspath(source_path)
+        abs_destination_base_path = os.path.abspath(destination_base_path)
+        abs_destination_path = os.path.join(abs_destination_base_path, destination_relative_path)
+
+        if not os.path.exists(abs_source_path):
+            return
+        if os.path.realpath(abs_source_path) == os.path.realpath(abs_destination_path):
+            return
+
+        final_abs_destination_base_path = get_directory(destination_base_path)
+        final_abs_destination_path = os.path.join(final_abs_destination_base_path, destination_relative_path)
+
+        replication_operation(abs_source_path, final_abs_destination_path)
+
+    if not application_log:
+        return
+
+    # there may be multiple log files, or the file name may be different due to zipping
+    logs = glob.glob('%s*' % application_log)
+    print_msg(
+        "Results of the build can be found in the temporary log file(s) %s" % ', '.join(logs),
+        log=_log,
+        silent=silent
+    )
+
+    if build_successful:
+        return
+
+    datetime_stamp = time.strftime("%Y%m%d") + '-' + time.strftime("%H%M%S")
+
+    errorlog_path = get_log_error_path(ec=easyconfig)
+    if errorlog_path is not None:
+        errorlog_path = os.path.join(errorlog_path, datetime_stamp)
+
+        for log_file in logs:
+            initialize_persistent_dir_and_replicate_if_paths_distinct(
+                copy_file,
+                get_log_dir,
+                log_file,
+                errorlog_path,
+                os.path.basename(log_file)
+            )
+
+        print_msg(
+            "Build logs of failed build copied to permanent storage: %s" % errorlog_path,
+            log=_log,
+            silent=silent
+        )
+
+    errorartifact_path = get_artifact_error_path(ec=easyconfig)
+    if errorartifact_path is not None:
+        errorartifact_path = os.path.join(errorartifact_path, datetime_stamp)
+
+        builddir = app.builddir
+        if is_readable(builddir):
+            initialize_persistent_dir_and_replicate_if_paths_distinct(
+                copy_dir,
+                get_artifact_dir,
+                builddir,
+                errorartifact_path,
+                app.get_relative_builddir_base_path()
```

Comment:
Similar here: `builddir == errorartifact_path / get_relative_builddir_base_path` can never be true, can it?